frontend/src/components/FindMatch.js

```diff
@@ -53,6 +55,10 @@ const FindMatch = () => {
     }
   }
 
+  window.onbeforeunload = (event) => {
+    deleteMatch(user._id, matchingSocket.id, difficulty) // reset matching service
+  }
+
```

Comment:
What you did has solved the self-matching bug but unfortunately does not solve the bug I experienced.

For my bug, I actually also deduced that we need to `deleteMatch` after the room is deleted but I am struggling to see where to put it. With your suggestion, I may have some idea of where I can put it.

I shall highlight the bug I experienced during our meeting tomorrow night.

But good catch, this PR can be merged in first.

Reply:
ohh let us know what bug u experience so we can create a new task to fix it separately